spacy/tokens/doc.pyx

```diff
@@ -1602,12 +1602,26 @@ cdef class Doc:
                 ents.append(char_span)
             self.ents = ents
 
-        # Add custom attributes. Note that only Doc extensions are currently considered, Token and Span extensions are
-        # not yet supported.
-        for attr in doc_json.get("_", {}):
+        # Add custom attributes for the whole doc
+        for attr in doc_json["_"]["docs"]:
```

Comment:
Will the index/subscript operator return an empty iterator if `docs` isn't found in `_` (or if `_` isn't found)? If not, we'll want to revert to using the `get` method with an empty dict as the default value.

Reply:
Yes, you're exactly right! Will be changed in the next commit